cmd/tsdb/main.go

```diff
@@ -53,8 +54,15 @@ func main() {
 		analyzePath          = analyzeCmd.Arg("db path", "database path (default is "+filepath.Join("benchout", "storage")+")").Default(filepath.Join("benchout", "storage")).String()
 		analyzeBlockID       = analyzeCmd.Arg("block id", "block to analyze (default is the last block)").String()
 		analyzeLimit         = analyzeCmd.Flag("limit", "how many items to show in each list").Default("20").Int()
+		dumpCmd              = cli.Command("dump", "dump samples from a TSDB")
+		dumpPath             = dumpCmd.Arg("db path", "database path (default is "+filepath.Join("benchout", "storage")+")").Default(filepath.Join("benchout", "storage")).String()
+		dumpMinTime          = dumpCmd.Flag("min-time", "minimum timestamp to dump").Default(strconv.FormatInt(math.MinInt64, 10)).Int64()
+		dumpMaxTime          = dumpCmd.Flag("max-time", "maximum timestamp to dump").Default(strconv.FormatInt(math.MaxInt64, 10)).Int64()
 	)
 
+	safeDBOptions := *tsdb.DefaultOptions
+	safeDBOptions.RetentionDuration = 0
```

Comment:
retention is not the only mutating operation so there is another PR that adds an option to open the db with all mutating operations disabled.

https://github.com/prometheus/tsdb/pull/517/files#diff-beaca4df72bb224a9c7b294c84b2b493R42

Reply:
I mainly cared about retention because I didn't want old data to disappear. I think I'll just keep this in and the other PR can rebase, or should I take it out?

Reply:
yep can keep it, I will make sure whichever one goes second makes the correct rebase changes.